test/suites/secondary_redis.bash

```diff
@@ -0,0 +1,95 @@
+SUITE_secondary_redis_PROBE() {
+     if [ "`which redis-server`" = "" ]; then
+        echo "no redis-server"
+        return
+    fi
+    if [ "`which redis-cli`" = "" ]; then
+        echo "no redis-cli"
+        return
+    fi
+}
+
+SUITE_secondary_redis_SETUP() {
+    unset CCACHE_NODIRECT
+    export CCACHE_SECONDARY_STORAGE="redis://localhost:7777"
+
+    generate_code 1 test.c
+}
+
+expect_number_of_cache_entries() {
+    local expected=$1
+    local host=$2
```

Comment:
Since redis v4 the `redis-cli` accepts an `-u <url>` command. Mabye you could expose that URL as part of the `_SETUP` function and avoid the `localhost 7777` arguments to `expect_number_of_cache_entries`.

Reply:
I think I will keep them as separate variables, for now. Maybe there should be some variables, though.

Reply:
Went with url after all, thanks for the suggestion!